lib/tasks/emailer.rake

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+namespace :emailer do
+  desc 'email task runner'
+  task :vouchers_to_merchants, [:time_range] => [:environment] do |_task, _args|
+    desc 'sends all voucher codes to all of their merchants'
+
+    ActiveRecord::Base.logger = Logger.new(STDOUT)
+    Rails.logger = Logger.new(STDOUT)
+
+    range = 1.year.ago
+    case _args.time_range
+    when 'month'
+      range = 1.month.ago
+    when 'week'
+      range = 1.week.ago
+    end
+
+    Seller.all.each do |seller|
+      unless seller.distributor
+        Rails.logger.warn("No distributor found for #{seller.seller_id}, skipping email...")
+        next
+      end
+
+      Rails.logger.info('Getting gift cards for ' + seller.seller_id)
+      html = '' # start building html with an empty string
+
+      html += "<h3> Hello #{seller.distributor.name} at #{seller.seller_id} </h3>"
+      html += "<p>Here's last #{_args.time_range}'s vouchers:</p>"
+
+      # query for all gift cards info for that seller
+      query = GiftCardDetail
+              .select(:gift_card_id, :value, :name, :email, :created_at, :expiration)
```

Comment:
I think you want seller_gift_card_id instead of gift_card_id

Reply:
Thanks for the catch!